src/worker/controller/actions/ethereum/CommitReceiptAction.js

```diff
@@ -13,63 +13,121 @@ class CommitReceiptAction {
     if (!task) return;
     let err = null;
 
-    try {
-      const txReceipt = await this._commitTask(task);
-      this._controller.logger().info(`[COMMIT_RECEIPT] success for task ${task.getTaskId()} receipt = ${txReceipt}`);
-    } catch (e) {
+    let result = await this._commitTask(task);
+    if (result.error) {
       this._controller.logger().error(`[COMMIT_RECEIPT] error for task ${task.getTaskId()} error=  ${e}`);
-      err = e;
+      err = result.error;
+    }
+    else {
+      this._controller.logger().info(`[COMMIT_RECEIPT] success for task ${task.getTaskId()} receipt = ${result.txReceipt}`);
     }
     if (callback) {
       callback(err);
     }
   }
-  _commitTask(task) {
-    if (task.getResult().isSuccess() && task.getResult().getOutput()) {
-      return this._commitSuccessTask(task);
-    } else if (task.getResult().isFailed()) {
-      return this._commitFailedTask(task);
+  async _commitTask(task) {
+    let res;
+    if (task.getResult().isSuccess()) {
+      res = await this._commitSuccessTask(task);
+    } else {
+      res = await this._commitFailedTask(task);
     }
-    throw new errors.TypeErr(`wrong type or missing fields in Result`);
+    return res;
   }
-  _commitFailedTask(task) {
+  async _commitFailedTask(task) {
```

Comment:
I saw you added `result` to many ipc messages, was wondering if the [failed task](https://github.com/enigmampc/enigma-p2p/blob/develop/docs/IPC_MESSAGES.md#failedtask-error-message) is also done accordingly, and if so would be great if you can point me to it :)

Reply:
Yes, look [here](https://github.com/enigmampc/enigma-p2p/blob/master/src/worker/controller/actions/tasks/ExecuteVerifiedAction.js#L54)